ruby/ql/lib/codeql/ruby/frameworks/core/Kernel.qll

```diff
@@ -197,4 +197,35 @@ module Kernel {
       super.getMethodName() = ["autoload", "autoload?"]
     }
   }
+
+  /**
+   * A call to `Array()`, that converts it's singular argument to an array.
+   * This summary is based on https://ruby-doc.org/3.2.1/Kernel.html#method-i-Array
+   */
+  private class KernelArraySummary extends SummarizedCallable {
+    KernelArraySummary() { this = "Array()" }
+
+    override MethodCall getACall() {
+      result.getMethodName() = "Array" and
+      // I have to have a simplified "KernelMethodCall" implementation inlined here, because relying on `UnknownMethodCall` results in non-monotonic recursion (even if using `getACall`).
+      (
+        result = API::getTopLevelMember("Kernel").getAMethodCall(_).asExpr().getExpr()
```

Comment:
Consider using something like `getAStaticArrayCall` here instead, in order to be able to go back to `getASimpleCall` (which means that the summary will also be used during type tracking).